<launch>
    <!-- start the realsense -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" >
        <arg name="device_type" value="d415"/>
        <arg name="filters" value="pointcloud"/>
    </include>

    <!-- visual perception -->
    <node name="processing_node" pkg="my_perception" type="perception_node" output="screen">
        <rosparam>
        cloud_topic: "/camera/depth/color/points"
        world_frame: "prbt_base"
        camera_frame: "camera_depth_frame"
        voxel_leaf_size: 0.001 <!-- mm -->
        x_filter_min: -2.5 <!-- m -->
        x_filter_max: 2.5 <!-- m -->
        y_filter_min: -2.5 <!-- m -->
        y_filter_max: 2.5 <!-- m -->
        z_filter_min: -2.5 <!-- m -->
        z_filter_max: 2.5 <!-- m -->
        plane_max_iterations: 100
        plane_distance_threshold: 0.03
        cluster_tolerance: 0.01
        cluster_min_size: 250
        cluster_max_size: 500000
        </rosparam>
    </node>
</launch>